pkg/controller/pxc/controller.go

```diff
@@ -969,7 +970,12 @@ func (r *ReconcilePerconaXtraDBCluster) deleteStatefulSet(cr *api.PerconaXtraDBC
 		return nil
 	}
 
-	err = r.client.Delete(context.TODO(), sfs.StatefulSet())
+	if !metav1.IsControlledBy(&sfsWithOwner, cr){
+		return nil
+	}
+
+	uid := sfsWithOwner.UID
+	err = r.client.Delete(context.TODO(), sfs.StatefulSet(), &client.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &uid}})
```

Comment:
here we also can pass just `&sfsWithOwner` instead of `sfs.StatefulSet()` just because they are the same.
Just to make code more readable and understandable, because now we are using `uid` from one object, and deleting another

Reply:
Updated accordingly